t/lib/t/MusicBrainz/Server/Edit/Medium/SetTrackLengths.pm

```diff
@@ -39,17 +49,32 @@ test 'Setting track lengths on medium with pregap track' => sub {
 
     MusicBrainz::Server::Test->prepare_test_database($c, '+tracklist');
 
+    # Note: Hidden pre-gap track is ignored at purpose, as pre-gap
+    # duration is considered as a maximum for hidden track length.
+
+    isa_ok exception {
+        $c->model('Edit')->create(
+            edit_type => $EDIT_SET_TRACK_LENGTHS,
+            editor_id => 1,
+            medium_id => 4,
+            cdtoc_id => 2
+        )
+    }, 'MusicBrainz::Server::Edit::Exceptions::NoChanges';
+
     my $edit = $c->model('Edit')->create(
         edit_type => $EDIT_SET_TRACK_LENGTHS,
         editor_id => 1,
-        medium_id => 4,
+        medium_id => 5,
         cdtoc_id => 2
     );
     isa_ok($edit => 'MusicBrainz::Server::Edit::Medium::SetTrackLengths');
 
-    my $medium = $c->model('Medium')->get_by_id(4);
+    my $medium = $c->model('Medium')->get_by_id(5);
     $c->model('Track')->load_for_mediums($medium);
+
+    # Hidden pre-gap track length is untouched at purpose, see above.
```

Comment:
Small grammar nitpick: "on purpose" is the right idiom.

Reply:
fixed in a0c35bd4444a00e3788a1ceccf2439250cfc138c and 33d0154b923d5c00caf524d473431cda5d511365